storage_service/locations/models/package.py

```diff
@@ -1112,9 +1141,34 @@ def finish_reingest(self, origin_location, origin_path, reingest_location, reing
             distutils.dir_util.copy_tree(reingest_metadata_dir,
                 original_metadata_dir)
 
+        # Replace preservation derivatives
+        reingest_objects_dir = os.path.join(reingest_full_path, 'data', 'objects')
+        original_objects_dir = os.path.join(path, 'data', 'objects')
+        preservation_regex = r'(.+)-\w{8}-\w{4}-\w{4}-\w{4}-\w{12}'
+        # Walk through all files
+        for dirpath, _, filepaths in os.walk(reingest_objects_dir):
+            for filepath in filepaths:
+                match = re.match(preservation_regex, filepath)
+                # If preservation file, copy and delete the old one
+                if match:
+                    reingest_preservation_path = os.path.join(dirpath, filepath)
+                    original_preservation_path = reingest_preservation_path.replace(reingest_objects_dir, original_objects_dir)
+                    # Check for another preservation derivative and delete
+                    dest_dir = os.path.dirname(original_preservation_path)
+                    dupe_preservation_regex = match.group(1) + r'-\w{8}-\w{4}-\w{4}-\w{4}-\w{12}'
+                    for p in os.listdir(dest_dir):
+                        if re.match(dupe_preservation_regex, p):
+                            del_path = os.path.join(dest_dir, p)
+                            LOGGER.info('Deleting %s', del_path)
+                            os.remove(del_path)
+                    # Copy new preservation derivative
+                    LOGGER.info('Moving %s to %s', reingest_preservation_path, original_preservation_path)
+                    shutil.copy2(reingest_preservation_path, original_preservation_path)
+
```

Comment:
While it's an edge case, I think this block will cause a bug if a transfer contains a file whose original name has a UUID suffix. For example, try to re-ingest a SIP whose original transfer contains a file named "Nemastylis_geminiflora_Flower-a464e051-b6db-46cf-9f45-64c00ae16171.PNG". I tried this and the resulting AIP (after re-ingest) did not contain the .tif derivative that was created during the original ingest.

Assuming I'm correct in identifying this as an error, maybe a better way of identifying preservation derivatives would be to look for files whose names are suffixed by UUIDs AND where there is another file with the same basename, minus the UUID suffix.